nemo/collections/asr/parts/utils/transcribe_utils.py

```
@@ -289,9 +290,22 @@ def write_transcription(
     else:
         pred_text_attr_name = 'pred_text'
 
+    if isinstance(transcriptions[0], rnnt_utils.Hypothesis):  # List[rnnt_utils.Hypothesis]
+        best_hyps = transcriptions
+        assert cfg.ctc_decoding.beam.return_best_hypothesis, "Works only for NBestHypothesis"
+    elif isinstance(transcriptions[0], list):  # List[List[rnnt_utils.Hypothesis]]
+        best_hyps, beams = [], []
+        for hyps in transcriptions:
+            best_hyps.append(hyps[0])
+            if not cfg.ctc_decoding.beam.return_best_hypothesis:
+                beam = []
+                for hyp in hyps:
+                    beam.append((hyp.text, hyp.score))
+                beams.append(beam)
+
```

else?  maybe throw an error?

---

raise TypeError

---

```
if not cfg.ctc_decoding.beam.return_best_hypothesis:
    ...
else:
    pass # it is fine, no error for 1 best option
```